### ISSUE TITLE
Empty string values not stored in Solr

### ISSUE THREAD

**Issue by peter721** [mongodb-labs/mongo-connector#212]
Having an issue using the mongo-connector against Solr with field values containing empty strings.

1. Mongo documents in the oplog contain sub-document with key "xxx".
2. Solr's schema is configured with a dynamic field specification like:

        <dynamicField name="xxx.*" type="string" indexed="true" stored="true"/>

Documents are stored in solr as I would expect, with all fields in the mongo sub-document flattened with the prefix "xxx.", but any field present in that sub-document that has an empty-string value is not subsequently present in the solr document - it appears to have been filtered out.

Perhaps it is a problem with my configuration. Any help you could give me is very much appreciated.

**Comment by peter721** [comment]
Hi, looks like part of my message got eaten last time. The dynamic field specification in solr's schema.xml looks like:
< dynamicField name="xxx.*" type="string" indexed="true" stored="true" />

Is there anybody that can help me with this issue?

**Comment by llvtt** [comment]
Hey @peter721 
It looks like `pysolr`, the Python interface to Solr that Mongo Connector uses, does not allow zero-length strings to be sent to Solr. See here: https://github.com/toastdriven/pysolr/blob/master/pysolr.py#L579-L586

It looks like the code for checking empty strings was added in April 2010, and there's a comment that notes "This should probably be removed when solved in core Solr level". I'm not sure what issue this was trying to circumvent, or if it's ok to remove this code now. I opened an [issue](https://github.com/toastdriven/pysolr/issues/150) on the Pysolr Github page to get more information on this.

In short: it's not possible to send empty-string values to Solr right now with Mongo Connector.

**Comment by atuljangra** [comment]
+1 having this issue.

**Comment by aherlihy** [comment]
@atuljangra Hello! Unfortunately the issue comes from PySolr and not mongo-connector, as addressed in [this issue] (https://github.com/toastdriven/pysolr/issues/150). There is nothing we can do until PySolr has been updated.
